web-app/js/Control.Select.js

```diff
@@ -85,13 +85,19 @@ L.Control.Select = L.Control.extend({
     },
 
     updateData:function(data){
-        var defval = this.defaultValue;
+        var defval = null;
         this.viewModel.facets.removeAll();
         for(var i in data){
-            this.viewModel.facets.push(new this.Model(data[i]))
+            var modelValue = new this.Model(data[i]);
+            this.viewModel.facets.push(modelValue);
+            if (data[i].name === this.options.defaultValue) {
+                defval = modelValue;
```

Comment:
you have to fire change event otherwise the map might not refresh. it is refreshing now since Map.loadColorByData calls colorBy.setValue. I think colorBy.setValue should be removed.